test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,558 @@
+				library.db.blocks.loadLastBlock = sinonSandbox
+					.stub()
+					.resolves(viewRow_full_blocks_list);
 
-			it('should call callback with error = null');
+				modules.blocks.lastBlock.set = sinonSandbox.spy();
 
-			it('should call callback with result containing the block');
+				blocksUtilsModule.loadLastBlock((err, block) => {
+					expect(block).to.be.an('object');
+					expect(block.id).to.equal('13068833527549895884');
+					expect(modules.blocks.lastBlock.set).to.have.been.calledWith(block);
+					done();
+				});
+			});
 		});
 	});
 
 	describe('getIdSequence', () => {
-		it('should call modules.blocks.lastBlock.get');
-
-		it('should call library.db.query');
-
-		it('should call library.db.query with sql.getIdSequence');
-
-		it(
-			'should call library.db.query with {height: height, limit: 5, delegates: constants.activeDelegates}'
-		);
-
-		describe('when db query fails', () => {
-			it('should call logger.error with error stack');
-
-			it('should call callback with the Blocks#getIdSequence error');
-		});
-
-		describe('when db query succeeds', () => {
-			describe('and returns no results', () => {
-				it('should call callback with an error');
+		it('should return error when library.db.blocks.getIdSequence fails', done => {
+			blocksUtilsModule.getIdSequence(10, (err, sequence) => {
+				expect(loggerStub.error.args[0][0]).to.contains(
+					"TypeError: Cannot read property 'length' of undefined"
+				);
+				expect(err).to.equal('Blocks#getIdSequence error');
+				expect(sequence).to.be.undefined;
+				done();
 			});
+		});
 
-			it(
-				'should add the genesis block to the end of the block array, if it does not contain it already'
-			);
-
-			it(
-				'should add the last block to the beginning of the block array, if it does not contain it already'
-			);
-
-			it('should call callback with error = null');
-
-			describe('result object', () => {
-				it('should assign firstHeight to the height of the last block');
+		it('should return error when no row is found', done => {
+			library.db.blocks.getIdSequence = sinonSandbox.stub().resolves([]);
 
-				it('should assign ids to a string of the block ids');
+			blocksUtilsModule.getIdSequence(10, (err, sequence) => {
+				expect(err).to.equal('Failed to get id sequence for height: 10');
+				expect(sequence).to.be.undefined;
+				done();
 			});
+		});
 
-			it('should call callback with result object');
+		it('should return valid block id list', done => {
+			library.db.blocks.getIdSequence = sinonSandbox
+				.stub()
+				.resolves([
+					{ id: 1, height: 2 },
+					{ id: 2, height: 3 },
+					{ id: 3, height: 4 },
+					{ id: 4, height: 5 },
+				]);
+
+			blocksUtilsModule.getIdSequence(10, (err, sequence) => {
+				expect(sequence).to.be.an('object');
+				expect(sequence.firstHeight).to.equal(1);
+				expect(sequence.ids).to.equal(
+					'9314232245035524467,1,2,3,4,6524861224470851795'
+				);
+				done();
+			});
 		});
```

Comment:
We should test more paths here:
- there are some paths related to genesis block
- there is a limit of 5
- lastBlock when not on list

Reply:
Ok thanks.

Reply:
Done.